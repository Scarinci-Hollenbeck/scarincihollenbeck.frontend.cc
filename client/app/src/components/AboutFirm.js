import React from 'react';
import { Link } from 'react-router-dom';

const AboutFirm = () => (
  <div className="w-100 mt-5">
    <div className="sidebar-title">
      Firm Resources
    </div>
    <div className="off-white">
      <ul className="pl-0 pt-2 pb-1 pr-1 sidebar-content-page">
        <li>
          <Link to='/category/firm-news' className="proxima-bold">
            Firm News
          </Link>
        </li>
        <li>
          <Link to='/category/law-firm-insights' className="proxima-bold">
            Firm Insights
          </Link>
        </li>
        <li>
          <Link to='/category/firm-events' className="proxima-bold">
            Firm Events
          </Link>
        </li>
      </ul>
    </div>
  </div>
);

export default AboutFirm;
