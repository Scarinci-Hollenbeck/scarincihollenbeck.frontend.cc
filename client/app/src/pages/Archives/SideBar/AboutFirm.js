import React from 'react';

const AboutFirm = () => (
  <div className="w-100 mt-5">
    <div className="sidebar-title">
      Firm Resources
    </div>
    <div className="off-white">
      <ul className="pl-0 pt-2 pb-1 pr-1 no-dots sidebar-content-page">
        <li>
          <a href="https://scarincihollenbeck/firm-news" className="small-excerpt">
            Firm News
          </a>
        </li>
        <li>
          <a href="https://scarincihollenbeck/law-firm-insights" className="small-excerpt">
            Firm Insights
          </a>
        </li>
        <li>
          <a href="https://scarincihollenbeck/firm-events" className="small-excerpt">
            Firm Events
          </a>
        </li>
      </ul>
    </div>
  </div>
);

export default AboutFirm;
