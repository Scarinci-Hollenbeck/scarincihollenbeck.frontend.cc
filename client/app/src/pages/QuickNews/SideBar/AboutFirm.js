import React from 'react';

const AboutFirm = () => (
  <div className="w-100 mt-5">
    <div className="sidebar-title">
      Firm Resources
    </div>
    <div className="off-white">
      <div className="p-2 list-style ml-45">
        <a href="https://scarincihollenbeck/firm-news" className="h5">
        Firm News
        </a>
      </div>
      <div className="p-2 pb-1 list-style ml-45">
        <a href="https://scarincihollenbeck/firm-events" className="h5">
        Firm Events
        </a>
      </div>
      <div className="p-2 pb-1 list-style ml-45">
        <a href="https://scarincihollenbeck/law-firm-insights" className="h5">
        Firm Insights
        </a>
      </div>
    </div>
  </div>
);

export default AboutFirm;
