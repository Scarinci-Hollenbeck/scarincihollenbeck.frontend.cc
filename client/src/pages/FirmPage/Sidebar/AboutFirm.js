import React from 'react';
import './index.scss';

const AboutFirm = () => (
  <div className="w-100 mt-3">
    <div className="sidebar-title">
      Firm Resources
    </div>
    <div className="off-white">
      <ul className="pl-0 pt-2 pb-1 pr-1 no-dots sidebar-content">
        <li>
          <a href={`${process.env.API_URL}/category/firm-news/`} className="text-capitalize small-excerpt">
            Firm News
          </a>
        </li>
        <li>
          <a href={`${process.env.API_URL}/category/firm-events/`} className="text-capitalize small-excerpt">
            Firm Events
          </a>
        </li>
        <li>
          <a href={`${process.env.API_URL}/category/law-firm-insights/`} className="text-capitalize small-excerpt">
          Firm Insights
          </a>
        </li>
      </ul>
    </div>
  </div>
);

export default AboutFirm;
