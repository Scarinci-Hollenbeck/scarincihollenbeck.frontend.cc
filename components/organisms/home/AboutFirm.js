import {
  AboutBlock,
  AboutBlocks,
  AboutCard,
  AboutCardContent,
  AboutCardDescription,
  AboutCardImage,
  AboutCardTitle,
  AboutDescription,
  AboutSection,
  AboutTitle,
} from 'styles/AboutFirm.style';
import { ContainerDefault } from 'styles/Containers.style';
import Image from 'next/image';
import LogoSeparator from 'components/common/LogoSeparator';
import { JSXWithDynamicLinks } from 'components/atoms/micro-templates/JSXWithDynamicLinks';

const AboutFirm = ({
  title,
  heroProfileLink,
  aboutHero,
  arcticle,
  heroPhoto,
}) => (
  <AboutSection>
    <ContainerDefault>
      <AboutBlocks>
        <AboutBlock>
          <AboutTitle>{title}</AboutTitle>
          <AboutDescription>
            <JSXWithDynamicLinks HTML={arcticle} />
          </AboutDescription>
        </AboutBlock>
        <LogoSeparator />
        <AboutBlock>
          <AboutCard>
            <AboutCardImage>
              <Image
                src={heroPhoto?.sourceUrl}
                width={280}
                height={374}
                alt={heroPhoto?.altText}
              />
            </AboutCardImage>
            <AboutCardContent>
              <AboutCardDescription>
                {aboutHero.split('\n').map((paragraph, index) => (
                  // eslint-disable-next-line react/no-array-index-key
                  <p key={`${aboutHero}-${index}`}>{paragraph}</p>
                ))}
              </AboutCardDescription>

              <AboutCardTitle href={heroProfileLink?.url}>
                {heroProfileLink?.title}
              </AboutCardTitle>
            </AboutCardContent>
          </AboutCard>
        </AboutBlock>
      </AboutBlocks>
    </ContainerDefault>
  </AboutSection>
);

export default AboutFirm;
