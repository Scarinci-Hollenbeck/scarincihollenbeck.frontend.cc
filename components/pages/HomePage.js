import Image from 'next/image';
import Link from 'next/link';
import Container from 'react-bootstrap/Container';
import HomeBanner from 'components/organisms/home/banner';
import HomeHoneyCombSection from 'components/organisms/home/honey-comb-section';
import HomeMainTag from 'components/organisms/home/main-tag';
import HomeSiteHead from 'components/shared/head/HomeSiteHead';
import { sortByKey } from 'utils/helpers';
import styles from 'styles/Home.module.css';
import marginStyles from 'styles/Margins.module.css';
import { CURRENT_DOMAIN } from 'utils/constants';
import dynamic from 'next/dynamic';

const HomeLocations = dynamic(() => import('components/organisms/home/locations'));
const HomeOurLeadership = dynamic(() => import('components/organisms/home/our-leadership'));
const HomeReviews = dynamic(() => import('components/organisms/home/reviews'));
const HomeWhoWeAreSection = dynamic(() => import('components/organisms/home/who-we-are-section'));
const HomePageLink = dynamic(() => import('components/organisms/home/page-link'));
const ArticleHero = dynamic(() => import('components/shared/ArticleHero'));

export default function HomePage({
  seo, posts, locations, leadership, awards,
}) {
  return (
    <>
      <HomeSiteHead
        title={seo.title}
        metaDescription={seo.metaDescription}
        canonicalUrl={CURRENT_DOMAIN}
      />
      <HomeBanner />
      <Container>
        <HomeMainTag />
        <HomeHoneyCombSection
          contentOne={(
            <Image
              src="/images/goalssh-400x400.png"
              alt="meet our attorneys"
              width={400}
              height={400}
              layout="intrinsic"
            />
          )}
          contentTwo={(
            <div className={`${styles.honeyCombContent} float-right`}>
              <p className={styles.honeycombTitle}>
                <strong>MEET OUR TEAM</strong>
              </p>
              <p>
                Our attorneys collaborate across the firm’s practice areas to achieve the best
                combination of knowledge, experience, and efficiency. We are dedicated to delivering
                outstanding client service.
              </p>
              <Link href="/attorneys">
                <a>Meet our attorneys</a>
              </Link>
            </div>
          )}
        />
        <HomeHoneyCombSection
          contentOne={(
            <div className={styles.honeyCombContent}>
              <p className={styles.honeycombTitle}>
                <strong>OUR SERVICES</strong>
              </p>
              <p>
                We help our clients achieve their goals by providing tailored services with the
                focused experience of a boutique firm by drawing upon the resources of the firm’s
                core practice areas.
              </p>
              <Link href="/practices">
                <a>See what we do</a>
              </Link>
            </div>
          )}
          contentTwo={(
            <div className="float-right">
              <Image
                src="/images/colabsh2-400x400.png"
                alt="meet our attorneys"
                width={400}
                height={400}
                layout="intrinsic"
              />
            </div>
          )}
        />
        <HomeWhoWeAreSection />
        <HomePageLink link="/firm-overview" title="More from our firm" />
        <HomeOurLeadership attorneys={leadership} />
        <HomeReviews awards={awards} />
        <HomePageLink link="/awards" title="Award Methodology" />
        <HomeLocations locations={sortByKey(locations.offices, 'id')} />
        <div className={marginStyles.mt6}>
          <ArticleHero content={posts} />
        </div>
        <HomePageLink
          link="/library/category/firm-news"
          title="Read more articles about our attorneys"
          margins="my-5"
        />
      </Container>
    </>
  );
}
