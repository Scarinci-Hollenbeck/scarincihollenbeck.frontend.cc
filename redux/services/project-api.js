import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

// Define a service using a base URL and expected endpoints
export const appApi = createApi({
  reducerPath: 'appApi',
  baseQuery: fetchBaseQuery({ baseUrl: '/api/' }),
  endpoints: (builder) => ({
    getAppApi: builder.query({
      query: (slug) => `${slug}`,
    }),
    appApi: builder.mutation({
      query: (request) => ({
        url: `${request.route}`,
        method: request?.method || 'POST',
        body: request?.payload,
      }),
    }),
  }),
});

// Export hooks for usage in functional components, which are
// auto-generated based on the defined endpoints
export const { useGetAppApiQuery, useAppApiMutation } = appApi;
