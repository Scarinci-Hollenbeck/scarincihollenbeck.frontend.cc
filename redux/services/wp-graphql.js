// Need to use the React-specific entry point to import createApi
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { GRAPHQL_API_URL } from '../../utils/constants';

// Define a service using a base URL and expected endpoints
export const wpGraphQl = createApi({
  reducerPath: 'graphqlApi',
  baseQuery: fetchBaseQuery({ baseUrl: GRAPHQL_API_URL }),
  endpoints: (builder) => ({
    getDataFromWPGraphQL: builder.mutation({
      query: (graphQLData) => ({
        url: '/',
        method: 'POST',
        body: graphQLData,
      }),
    }),
  }),
});

// Export hooks for usage in functional components, which are
// auto-generated based on the defined endpoints
export const { useGetDataFromWPGraphQLQuery } = wpGraphQl;
