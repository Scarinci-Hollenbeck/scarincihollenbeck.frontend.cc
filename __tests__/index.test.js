import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { RouterContext } from 'next/dist/shared/lib/router-context.shared-runtime';
import mockRouter from 'next-router-mock';
import renderer from 'react-test-renderer';
import HomePage from '../components/pages/HomePage';
import { wpGraphQl } from '../redux/services/wp-graphql';
import { appApi } from '../redux/services/project-api';
import attorneysReducer from '../redux/slices/attorneys.slice';
import formsReducer from '../redux/slices/forms.slice';
import sizesReducer from '../redux/slices/sizes.slice';
import AboutFirm from '../components/organisms/home/AboutFirm';

// Mocking next/dynamic
jest.mock('next/dynamic', () => (component) => function DynamicComponent() {
  return (
    <section data-testid={component.displayName || 'dynamic-component'} />
  );
});

// Set up the store
const store = configureStore({
  reducer: {
    [wpGraphQl.reducerPath]: wpGraphQl.reducer,
    [appApi.reducerPath]: appApi.reducer,
    attorneys: attorneysReducer,
    forms: formsReducer,
    sizes: sizesReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(appApi.middleware),
});

// Tests
describe('HomePage', () => {
  it('The Home banner renders', () => {
    render(
      <Provider store={store}>
        <RouterContext.Provider value={mockRouter}>
          <HomePage />
        </RouterContext.Provider>
      </Provider>,
    );

    const homeBanner = screen.getByTestId('home-banner');
    expect(homeBanner).toBeInTheDocument();
  });

  it('The main title renders correctly', () => {
    const tree = renderer
      .create(<h1>Decades of Legal Experience with Recognized Results</h1>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('The banner info cards render', () => {
    render(
      <Provider store={store}>
        <RouterContext.Provider value={mockRouter}>
          <HomePage />
        </RouterContext.Provider>
      </Provider>,
    );
    const bannerCards = screen.getByTestId('banner-cards');
    expect(bannerCards).toBeInTheDocument();
  });

  it('"Who we are" section renders', () => {
    render(
      <AboutFirm
        title="About us"
        heroProfileLink={{ title: 'link', url: '/' }}
        aboutHero="Super hero - super skill"
        arcticle="Frog"
        heroPhoto="/images/no-image-found-diamond.png"
      />,
    );
    const whoWeAreSection = screen.getByTestId('who-we-are');
    expect(whoWeAreSection).toBeInTheDocument();
  });
});
