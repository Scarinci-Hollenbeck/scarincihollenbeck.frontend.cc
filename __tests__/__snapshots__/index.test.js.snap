// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage The main title renders correctly 1`] = `
<h1>
  Decades of Legal Experience with Recognized Results
</h1>
`;
